### DISCUSSION TITLE
PDF Image rendering on GitHub Pages, breaks when using image in a subdirectory

### DISCUSSION THREAD

**Discussion by s2t2** [quarto-dev/quarto-cli#10471] [Q&A] [answered]
### Description

I am deploying a quarto book project to github pages. The book renders locally, but the GitHub Pages build is failing.

The book references an image in the sidebar logo configuration option, as well as in a QMD file.

When the image is in the same directory as the QMD file ("./my_img.jpg"), it works, but it doesn't work when the image is nested within a subdirectory (like "./images/my_img.jpeg")

See working example, stopped working once image moved to subdirectory:
https://github.com/s2t2/quarto-book-pdf-download-test/

https://github.com/s2t2/quarto-book-pdf-download-test/commits/main/

After it stopped working, I attempted to change the image path, but no luck.

See original conversation: https://github.com/quarto-dev/quarto-cli/issues/10468

**Comment by mcanouil** [comment]
There is no issue with sub directory: https://github.com/mcanouil/book-pdf-download

**Reply by mcanouil** [reply, marked as answer]
Look at the extensions of your images.

<img width="300" alt="image" src="https://github.com/user-attachments/assets/632bb5e7-119f-4171-8a6a-49401aa0c03a">

**Comment by s2t2** [comment]
Interesting. Thanks for all your help by the way. 

I wonder why the original build in the original conversation is failing then. I had been trying to pinpoint the issue but I guess image in subdirectory wasn't the issue.

The issue seems to be in the PDF generation process

```
Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages
ERROR: 
compilation failed- error
Unable to load picture or PDF file './images/python-banner.jpeg'.
<to be read again> 
                   }
l.237 ...ludegraphics{./images/python-banner.jpeg}
```

https://github.com/prof-rossetti/intro-software-dev-python-book/actions/runs/10256503671/job/28375700196

**Reply by mcanouil** [reply]
The file is not found, so check that you spell the file paths properly including extension.
Note that, it should fail the exact same way locally.

**Comment by s2t2** [comment]
From what I can see, the image files exists and are properly linked to. The images given their current filepaths resolve for the website locally, and the PDF locally, and the website on GitHub Pages. Just not for PDF on GitHub Pages.

The PDF renders locally, without errors:

```
Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023/Homebrew) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
running xelatex - 2
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023/Homebrew) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
running xelatex - 3
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023/Homebrew) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  

Output created: _build/index.html
```


<img width="552" alt="Screenshot 2024-08-05 at 8 01 23 PM" src="https://github.com/user-attachments/assets/bd2ca25d-fb77-4e9f-8653-66f6229df489">



Super stumped.

**Reply by mcanouil** [reply]
Could you try to remove everything unrelated to the issue in a dedicated branch?
Remove the pieces one by one until you are left only with the source of the issue.
Doing this process usually allows to figure out what was wrong.

Also, you are still using outdated actions.
You should really take into accounts the messages from your CI/CD.
<img width="1122" alt="image" src="https://github.com/user-attachments/assets/5af00c30-1d87-428b-b8f1-37f7cb3e7bab">

**Comment by s2t2** [comment]
I updated some image paths in the original book, from `./images/my_image.jpeg` to `/images/my_image.jpeg`, and that seemed to resolve the build, which in turn also resolves the pdf- download issue, because now the PDF is available after the build has completed.

I appreciate all of your support throughout this process 🙌

**Reply by mcanouil** [reply]
Glad you solved your issue.

**Comment by abdulwasay1100** [comment]
I'm also facing similar kind of issue but still confuse how to resolve it?!


**Description**

I'm working on a mk_docs project now I want to convert some SOP.qmd files into pdf 
so I also added quarto.yml file to make pdfs consistant and SOP files contain images with paths like this: "../../../img/DMP/Anmeldung%20Adminbereich.png"

in the quarto.yml file I mentioned the resource path which is "docs\img" and the img dir also contains sub-dirs.
when serving mkdocs locally it rendering the images from the same path but when converting into pdfs it gives below error:

[[ Error creating PDF for merged_sops\sop-dmp-datapub_merged.qmd: pandoc --output sop-dmp-datapub_merged.tex
  to: latex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  toc: true
  number-sections: true
  
metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - \KOMAoption{captions}{tableheading}
  block-headings: true
  resources:
    - ../docs/img
  

Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages
ERROR: 
compilation failed- error
Unable to load picture or PDF file '../../../img/DMP/DMP_recommendedMetadata_Keywords_TypeOfData.png'.
<to be read again> 
                   }
l.351 ...mmendedMetadata_Keywords_TypeOfData.png}}    ]].


with short path it works but affecting my project nav file structure.

how can I resolve this issue without affecting the file nav structure?!
